mapfishapp/src/main/webapp/app/js/GEOR_Querier.js

```diff
@@ -0,0 +1,320 @@
+                }
+            } else if (!(f.value && f.type && 
+                (f.property || f.CLASS_NAME == "OpenLayers.Filter.Spatial"))) {
+                GEOR.util.infoDialog({
+                    msg: tr("Fields of filters with a red mark are mandatory")
+                });
+                return false;
+            }
+        }
+        return true;
+    },
+
+
+    /**
+     * Method: search
+     * Gets the Filter Encoding string and sends the getFeature request
+     */
+    search: function() {
+        // we quickly check if nothing lacks in filter
+        var filter = this.filterbuilder.getFilter();
+        if (!this._checkFilter(filter)) {
+            return;
+        }
+
+        this.fireEvent("search", {
+            // TODO: tell on which layerRecord the search is running ?
+            html: tr("<div>Searching...</div>")
+        });
+
+        // we deactivate draw controls before the request is done.
+        this.filterbuilder.deactivateControls();
+
+        // we need to pass the geometry name at protocol creation, 
+        // so that the format has the correct geometryName too.
+        GEOR.ows.WFSProtocol(this.pseudoRecord, this.map, {
+                geometryName: this.geometryName
+        }).read({
+            maxFeatures: GEOR.config.MAX_FEATURES,
+            // some mapserver versions require that we list all fields to return 
+            // (as seen with 5.6.1):
+            // see http://applis-bretagne.fr/redmine/issues/1996
+            propertyNames: this.layerFields.concat(this.geometryName) || [],
+            filter: filter,
+            callback: function(response) {
+                // Houston, we've got a pb ...
+                
+                // When using WFS 1.0, getFeature requests are honored with geodata in the original data's SRS 
+                // ... which might not be the map SRS !
+                // When using GeoServer, an additional parameter can be used (srsName) to ask for feature reprojection, 
+                // which is not part of the WFS 1.0 spec
+                // When using MapServer, it seems that we have no way to do the same.
+                
+                // So, basically, we have two solutions :
+                // - stick to WFS 1.1 spec, which leaves WFS servers with old MapServers unqueryable. 
+                //   We don't want that.
+                // - stick to WFS 1.0 spec and add the srsName parameter (in case we're in front of a GeoServer). 
+                // In that case, WFSes made with MapServer will return geometries in the original data SRS ... 
+                // which could be parsed, and reprojected on the client side using proj4js when user agrees to do so
+                
+                if (!response.success()) {
+                    return;
+                }
+                
+                var model = (this.attributeStore.getCount() > 0) ? 
+                    new GEOR.FeatureDataModel({
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["GEOR" is not defined.](https://www.codacy.com/app/georchestra/georchestra/file/2337677315/issues/source?bid=3200181&fileBranchId=3209168#l297)